receiver/statsdreceiver/protocol/statsd_parser.go

```diff
@@ -48,23 +55,72 @@ type statsDMetric struct {
 	labelValues      []*metricspb.LabelValue
 }
 
+func (p *StatsDParser) Initialize() error {
+	gauges = make(map[string]*metricspb.Metric)
+	counters = make(map[string]*metricspb.Metric)
```

Comment:
so this might be that I have not written any real go code in a year, but should this be:

```
p.gauges = make(map[string]*metricspb.Metric)
p.counters = make(map[string]*metricspb.Metric)
```

??

Or is this some go magic that I am unaware of

Reply:
Oh, `gauges` and `counters` are defined outside of the func. you could treat them as global variables. I did not put them as fields of the StatsDParser struct so no need in this case.

Reply:
Please add these fields to a struct. There should be comments somewhere around here about the synchronization story. I believe we discussed the use of a channel, but I haven't finished reviewing the PR yet and haven't seen it.